kernel/thread.c

```diff
@@ -648,6 +649,75 @@ char *z_setup_new_thread(struct k_thread *new_thread,
 	return stack_ptr;
 }
 
+#ifdef CONFIG_VALIDATE_THREAD_STACK_POINTER
+static bool is_stack_in_section(k_thread_stack_t *stack_base, size_t stack_size,
```

Comment:
Just for clarification, validating that a pointer to physical memory lies within a given region can literally be done anywhere, does not need to be done in kernel context, and does not require any special permissions. It's just doing math using a non-constant input and two probably-compile-time-const inputs.

This could be a static inline function in e.g. kernel.h, or it could just be a part of the application.